examples/ShowPlayer.cpp

```diff
@@ -85,55 +92,84 @@ int ShowPlayer::Playback(unsigned int iterations,
   return ola::EXIT_OK;
 }
 
-void ShowPlayer::SendNextFrame() {
-  DmxBuffer buffer;
-  unsigned int universe;
-  ShowLoader::State state = m_loader.NextFrame(&universe, &buffer);
-  switch (state) {
-    case ShowLoader::END_OF_FILE:
-      HandleEndOfFile();
-      return;
-    case ShowLoader::INVALID_LINE:
-      m_client.GetSelectServer()->Terminate();
-      return;
-    default:
-      {}
-  }
 
-  state = RegisterNextTimeout();
+/**
+ * Start playback
+ */
+void ShowPlayer::Start() {
+  SeekTo(m_start);
+}
+
 
-  OLA_INFO << "Universe: " << universe << ": " << buffer.ToString();
-  ola::client::SendDMXArgs args;
-  m_client.GetClient()->SendDMX(universe, buffer, args);
-
-  switch (state) {
-    case ShowLoader::END_OF_FILE:
-      HandleEndOfFile();
-      return;
-    case ShowLoader::INVALID_LINE:
-      m_client.GetSelectServer()->Terminate();
-      return;
-    default:
-      {}
+/**
+ * Seek to @p time in the show file
+ * @param time the time (in milliseconds) to seek to
+ */
+void ShowPlayer::SeekTo(const unsigned int time) {
+  // Seeking to a time before the playhead's position requires moving from the
+  // beginning of the file.  This could be optimized more if this happens
+  // frequently.
+  if (time < m_playback_pos) {
+    m_loader.Reset();
+    m_playback_pos = 0;
   }
+
+  // Keep reading through the show file until desired time is reached.
+  ShowEntry entry;
+  unsigned int playhead_time = m_playback_pos;
+  do {
+    ShowLoader::State state = m_loader.NextEntry(&entry);
+    switch (state) {
+      case ShowLoader::END_OF_FILE:
+        HandleEndOfFile();
+        return;
+      case ShowLoader::INVALID_LINE:
+        m_client.GetSelectServer()->Terminate();
+        return;
+      default: {}
+    }
+    playhead_time += entry.next_wait;
+  } while (playhead_time < time);
+  // Adjust the timeout to handle landing in the middle of the entry's timeout
+  m_playback_pos = time;
+  entry.next_wait = playhead_time - time;
+  SendFrame(entry);
 }
 
 
 /**
- * Get the next time offset
+ * Send the next frame in the show file
  */
-ShowLoader::State ShowPlayer::RegisterNextTimeout() {
-  unsigned int timeout;
-  ShowLoader::State state = m_loader.NextTimeout(&timeout);
-  if (state != ShowLoader::OK) {
-    return state;
+void ShowPlayer::SendNextFrame() {
+  ShowEntry entry;
+  ShowLoader::State state = m_loader.NextEntry(&entry);
+  if (state == ShowLoader::END_OF_FILE || m_playback_pos >= m_stop) {
+    HandleEndOfFile();
+    return;
+  } else if (state == ShowLoader::INVALID_LINE) {
+    m_client.GetSelectServer()->Terminate();
+    return;
   }
+  SendFrame(entry);
+}
 
-  OLA_INFO << "Registering timeout for " << timeout << "ms";
+
+/**
+ * Send @p frame and wait for next
```

Comment:
Does the Doxygen not want to be `@p entry` from looking at https://www.doxygen.nl/manual/commands.html#cmdp

Reply:
Yup, that's what I get for trusting CLion's rename tools...

Reply:
Perhaps it's time for a bug report...